bbq/compiler/compiler.go

```diff
@@ -575,6 +575,18 @@ func (c *Compiler[_]) VisitContinueStatement(_ *ast.ContinueStatement) (_ struct
 }
 
 func (c *Compiler[_]) VisitIfStatement(statement *ast.IfStatement) (_ struct{}) {
+
+	// If statements can be coming from inherited conditions.
+	// If so, use the corresponding elaboration.
+	stmtElaboration, ok := c.ExtendedElaboration.conditionsElaborations[statement]
+	if ok {
+		prevElaboration := c.ExtendedElaboration
+		c.ExtendedElaboration = stmtElaboration
+		defer func() {
+			c.ExtendedElaboration = prevElaboration
+		}()
+	}
```

Comment:
Maybe we can extract this into a "wrapper" function which takes a function that is invoked while the extended elaboration is effective, e.g. something like 

```go
func (c *Compiler[_]) withConditionExtendedElaboration(statement ast.Statement, f func()) {
	stmtElaboration, ok := c.ExtendedElaboration.conditionsElaborations[statement]
	if ok {
		prevElaboration := c.ExtendedElaboration
		c.ExtendedElaboration = stmtElaboration
		defer func() {
			c.ExtendedElaboration = prevElaboration
		}()
	}
	f()
}
```

Reply:
Good idea! Added in 48c8488da8a147e455775bebe77bc6396a63bf97